python/fusion_engine_client/analysis/data_loader.py

```diff
@@ -172,7 +174,10 @@ def open(self, path, save_index=True, ignore_index=False):
         self.close()
 
         self.reader = MixedLogReader(input_file=path, save_index=save_index, ignore_index=ignore_index,
-                                     return_bytes=True, return_message_index=True)
+                                     return_bytes=True, return_message_index=True, source_ids=self.source_ids)
+
+        # Narrow down source IDs to the ones that are requested AND are available.
+        self.source_ids = self.reader.requested_source_ids
```

Comment:
Code and comment don't seem to match. Is that because the code is non-obvious? or because the comment is incorrect?

Looking a little deeper, it looks like the code is not obvious. On instantiation, the `reader` is going to populate `requested_source_ids` not purely from a request but from the intersection of a request and what is actually available. That might be worth clarifying. If you can think of a way to clarify this behavior by way of clever naming, rather than a comment, that is preferable.

Reply:
We should not be passing `source_ids` to the log reader constructor in this class. We are treating it the same way we do time range and message types, so it should be passed to `read()` only